### PR TITLE
Add Git Tags table for azure-devops plugin

### PR DESCRIPTION
**Pull request by Balfa**
## Hey, I just made a Pull Request!

Added a new Git Tags tab (based on the existing Pull Requests one) for the **azure-devops** plugin.
It contains similar information to the [git tags page](https://docs.microsoft.com/en-us/azure/devops/repos/git/git-tags?view=azure-devops&tabs=browser#view-tags-in-the-tags-view) on Azure Devops itself.

image

#### Testing locally
1. Configure your `azureDevOps` section in `app-config.yaml` with appropriate credentials
2. In `plugins/github-actions/examples/sample.yaml`, replace the annotations section with the following:
   ```yaml
   annotations:
     dev.azure.com/project-repo: AnyProjectInYourAdoOrg/AnyRepoInThatProject
   ```
3. When you visit the backstage entity, you should see tags for the repo supplied above.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
Hi @Balfa, thanks for your contribution! This is cool 😎

I'll do a full review on Friday. I'm off today and I know I'll be busy catching up tomorrow

File: plugins/azure-devops/src/components/GitTagTable/GitTagTable.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+];
+
+export const GitTagTable = () => {
+  const { entity } = useEntity();
+
+  const { items, loading, error } = useGitTags(entity);
+
+  if (error) {
+    return (
+      <div>
```
Comment (freben) [inline plugins/azure-devops/src/components/GitTagTable/GitTagTable.tsx:68]: probably not a necessary div

**Review by freben — APPROVED** [review]
Looks good to me at least :)

File: plugins/azure-devops/src/components/GitTagTable/GitTagTable.tsx (hunk 1 above)
Comment (Balfa) [reply plugins/azure-devops/src/components/GitTagTable/GitTagTable.tsx:68]: Possibly. This is just copied from [PullRequestTable.tsx](https://github.com/backstage/backstage/blob/02e35fe7688c47e49b785ad4ce643c1bf0ae86bc/plugins/azure-devops/src/components/PullRequestTable/PullRequestTable.tsx#L107). Hope you don't mind if I leave it this way for now at least to keep consistency?

**Comment by Balfa** [comment]
> Hi @Balfa, thanks for your contribution! This is cool 😎
> 
> I'll do a full review on Friday. I'm off today and I know I'll be busy catching up tomorrow

@awanlin , freben has just approved this. Is this instead of your review, or are you still wanting to put your review on it? :)

**Comment by freben** [comment]
Nah he's at a different company but has good knowledge of both azure as a system (which i don't) and the plugin itself, so i'll appreciate his stamp on it as well

**Comment by Balfa** [comment]
> 

Thanks @awanlin. Appreciate your diligence. Looking forward to feedback  😊

File: plugins/azure-devops/README.md [hunk 2]
```diff
@@ -16,6 +16,12 @@ Lists the top _n_ Active, Completed, or Abandoned Pull Requests for a given repo
 
 ![Azure Repos Pull Requests Example](./docs/azure-devops-pull-requests.png)
 
+### Azure Git Tags
+
+Lists all Git Tags for a given repository
+
+![Azure Repos Pull Requests Example](./docs/azure-devops-git-tags.png)
```
Comment (awanlin) [inline plugins/azure-devops/README.md:23]: "Pull Requests" should be changed to "Git Tags"

File: plugins/azure-devops/README.md [hunk 3]
```diff
@@ -16,6 +16,12 @@ Lists the top _n_ Active, Completed, or Abandoned Pull Requests for a given repo
 
 ![Azure Repos Pull Requests Example](./docs/azure-devops-pull-requests.png)
 
+### Azure Git Tags
```
Comment (awanlin) [inline plugins/azure-devops/README.md:19]: This should be "Azure Repos Git Tags" seeing as Git Tags is a feature of Azure Repos

File: plugins/azure-devops/README.md [hunk 4]
```diff
@@ -155,6 +161,42 @@ To get the Azure Repos component working you'll need to do the following two ste
...
+       <EntityLayout.Route if={isAzureDevOpsAvailable} path="/git-tags" title="Git Tags">
+         <EntityAzureGitTagsContent />
+       </EntityLayout.Route>
+       // ...
+     </EntityLayout>
+   ```
+
+**Notes:**
+
+- You'll need to add the `EntityLayout.Route` above from step 2 to all the entity sections you want to see Git Tags in. For example if you wanted to see Pull Requests when looking at Website entities then you would need to add this to the `websiteEntityPage` section.
```
Comment (awanlin) [inline plugins/azure-devops/README.md:197]: "Pull Requests" should be "Git Tags"

File: plugins/azure-devops/src/components/GitTagTable/GitTagTable.tsx (hunk 1 above)
Comment (awanlin) [reply plugins/azure-devops/src/components/GitTagTable/GitTagTable.tsx:68]: Could pull this out and do the same in the `PullRequestTable.tsx` if you don't mind? It's not needed at all

**Review by awanlin — CHANGES_REQUESTED** [review]
Really good work! All of my comments are pretty small text changes.

When I ran this against our on-prem instance everything worked great. One item I did notice is that when tags are created by a Service Account there is no commit. Not something we need to deal with now but might be nice to add on as a feature if the data is there.

File: plugins/azure-devops/README.md (hunk 2 above)
Comment (Balfa) [reply plugins/azure-devops/README.md]: Thanks for the feedback, and sorry for the whoopsies! I've pushed changes. Let me know if this resolves your comments to your satisfaction!

**Review by awanlin — APPROVED** [review]
Awesome, excited to see this fully in action, thanks again for contributing this and looking forward to anything else you work on!

**Review by freben — APPROVED** [review]

**Pull request merged**